pkg/controlplane/control/manager.go

```diff
@@ -43,28 +53,106 @@ func (e exportServiceNotExistError) Error() string {
 		e.name.Namespace, e.name.Name)
 }
 
+type conflictingServiceError struct {
+	service *v1.Service
```

Comment:
you coul just copy the `NamespacedName` and `ManagedBy` label and then let the rest of the Service be gc'ed